classes/PHPTAL/StringSource.php

```diff
@@ -20,7 +20,13 @@
 class PHPTAL_StringSource implements PHPTAL_Source
 {
     const NO_PATH_PREFIX = '<string ';
+    private string $_data;
+    private string $_realpath;
 
+    /**
+     * @param string $data
```

Comment:
Would be best to use a native type hint, to avoid a crash during property assignment